File: android/app/src/main/java/com/vaultsync/domain/usecase/CalculateFileHashUseCase.kt
package com.vaultsync.domain.usecase

import com.vaultsync.domain.repository.CryptoService

class CalculateFileHashUseCase(
    private val cryptoService: CryptoService
) {
    suspend operator fun invoke(data: ByteArray): String {
        return cryptoService.calculateSha256(data)
    }
}

File: ios/Sources/VaultSync/Domain/UseCases/CalculateFileHashUseCase.swift
import Foundation

public final class CalculateFileHashUseCase: Sendable {
    private let cryptoService: any CryptoService

    public init(cryptoService: any CryptoService) {
        self.cryptoService = cryptoService
    }

    public func execute(data: Data) async -> String {
        await cryptoService.calculateSha256(data: data)
    }
}
